src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java

```diff
@@ -47,211 +41,167 @@
  */
 public class PluginCompatTesterConfig {
 
-    private static final Logger LOGGER = Logger.getLogger(PluginCompatTesterConfig.class.getName());
-
-    // A working directory where the tested plugin's sources will be checked out
-    public File workDirectory;
-
-    // Path for maven settings file where repository will be provided allowing to
-    // download jenkins-core artifact (and dependencies)
-    private File m2SettingsFile;
-
     // The megawar
-    private File war;
+    @NonNull private final File war;
 
-    @CheckForNull private File externalMaven;
+    // A working directory where the tested plugin's sources will be checked out
+    @NonNull private final File workingDir;
 
     // List of plugin artifact ids on which tests will be performed
-    // If null, tests will be performed on every plugins retrieved from update center
-    private List<String> includePlugins;
+    // If empty, tests will be performed on every plugins retrieved from update center
+    @NonNull private List<String> includePlugins = List.of();
```

Comment:
this breaks the [ability to test all plugins](https://github.com/jenkinsci/plugin-compat-tester/blob/7c9c23be7c85067d05ec87819beb8fcf18a854e6/src/main/java/org/jenkins/tools/test/PluginCompatTester.java#L141-L145)

the code to check if something is included is still looking for null, and so now if you do not specify any included plugins then no plugins are tested.  
(thus --exclude-plugins ability is also broken)

there could be some others that need adapting.

I would have expected spotbugs to say redundant null check of non null value but well....

Reply:
Thanks, fixed up as of commit 4c5608c